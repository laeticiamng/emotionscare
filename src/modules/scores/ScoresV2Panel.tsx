// @ts-nocheck
import React from 'react';

export default function ScoresV2Panel() {
  return (
    <div className="p-4">
      <p className="text-muted-foreground">Panel de visualisation des scores - En d√©veloppement</p>
    </div>
  );
}
