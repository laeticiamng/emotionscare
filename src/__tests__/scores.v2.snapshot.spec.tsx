import React from "react";
import { describe, it, expect, vi, beforeEach } from "vitest";
import { render } from "@testing-library/react";
import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
import ScoresV2Panel from "@/app/modules/scores/ScoresV2Panel";

const mockDashboardData = vi.hoisted(() => ({
  moodTrend: [
    { date: "2024-05-01", mood: 7.2, energy: 6.8, annotation: "Journal du matin" },
    { date: "2024-05-02", mood: 7.8, energy: 7.1, annotation: "Respiration 4-7-8" },
    { date: "2024-05-03", mood: 8.0, energy: 7.5, annotation: "Session adaptive music" },
  ],
  weeklySessions: [
    { week: "S18", guided: 3, breathwork: 2, vr: 1, journaling: 2, total: 8 },
    { week: "S19", guided: 4, breathwork: 3, vr: 2, journaling: 2, total: 11 },
  ],
  heatmap: [
    { day: "Lun", slot: "Matin", intensity: 72, dominantMood: "Positif", sessions: 2 },
    { day: "Mar", slot: "Après-midi", intensity: 65, dominantMood: "Créatif", sessions: 1 },
    { day: "Jeu", slot: "Soir", intensity: 78, dominantMood: "Serein", sessions: 2 },
    { day: "Sam", slot: "Midi", intensity: 54, dominantMood: "Curieux", sessions: 1 },
  ],
  summary: {
    moodAverage: 7.7,
    moodVariation: 0.8,
    bestMoodDay: { date: "2024-05-03", mood: 8.0, energy: 7.5, annotation: "Session adaptive music" },
    sessionsAverage: 9.5,
    lastWeek: { week: "S19", guided: 4, breathwork: 3, vr: 2, journaling: 2, total: 11 },
    level: 4,
    currentExperience: 1680,
    nextLevelExperience: 2200,
    levelProgress: 76,
    streakDays: 6,
    mostIntenseSlot: { day: "Jeu", slot: "Soir", intensity: 78, dominantMood: "Serein", sessions: 2 },
    totalSessions: 19,
  },
  source: "remote" as const,
}));

vi.mock("@/contexts/AuthContext", () => ({
  useAuth: () => ({
    user: { id: "user-123", email: "demo@example.com" },
    isAuthenticated: true,
  }),
}));

vi.mock("@/services/scoresDashboard.service", () => ({
  fetchScoresDashboard: vi.fn().mockResolvedValue(mockDashboardData),
  SCORES_DASHBOARD_FALLBACK: mockDashboardData,
}));

vi.stubGlobal(
  "ResizeObserver",
  class {
    observe() {}
    unobserve() {}
    disconnect() {}
  },
);

const queryClient = new QueryClient({
  defaultOptions: {
    queries: { retry: false },
    mutations: { retry: false },
  },
});

const wrapper = ({ children }: { children: React.ReactNode }) => (
  <QueryClientProvider client={queryClient}>{children}</QueryClientProvider>
);

beforeEach(() => {
  queryClient.clear();
});

describe("ScoresV2Panel", () => {
  it("rend un panneau scores", () => {
    const { container, getByText } = render(<ScoresV2Panel />, { wrapper });
    expect(getByText(/Scores/i)).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
