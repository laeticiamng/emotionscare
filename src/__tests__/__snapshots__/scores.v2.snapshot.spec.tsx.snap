// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ScoresV2Panel > rend un panneau scores 1`] = `
<div>
  <section
    aria-label="Scores V2"
    class="space-y-8"
  >
    <div
      class="flex flex-col gap-4 md:flex-row md:items-center md:justify-between"
    >
      <div
        class="space-y-1"
      >
        <h1
          class="text-3xl font-semibold tracking-tight"
        >
          Scores
        </h1>
        <p
          class="text-muted-foreground"
        >
          Progression, streaks et badges
        </p>
        <p
          class="text-xs text-muted-foreground"
        >
          Synchronisé en temps réel avec vos activités Supabase.
        </p>
      </div>
      <div
        class="flex flex-wrap items-center gap-3 text-sm"
      >
        <div
          aria-label="Niveau 4"
          class="inline-flex items-center rounded-md border text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent bg-secondary text-secondary-foreground hover:bg-secondary/80 px-3 py-1"
        >
          Niveau 
          4
        </div>
        <div
          aria-label="6 jours de suite"
          class="flex items-center gap-1 text-muted-foreground"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-flame h-4 w-4 text-orange-500"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M8.5 14.5A2.5 2.5 0 0 0 11 12c0-1.38-.5-2-1-3-1.072-2.143-.224-4.054 2-6 .5 2.5 2 4.9 4 6.5 2 1.6 3 3.5 3 5.5a7 7 0 1 1-14 0c0-1.153.433-2.294 1-3a2.5 2.5 0 0 0 2.5 2.5z"
            />
          </svg>
          <span>
            6
             j de suite
          </span>
        </div>
        <div
          class="flex items-center gap-1 text-muted-foreground"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-calendar-range h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              height="18"
              rx="2"
              width="18"
              x="3"
              y="4"
            />
            <path
              d="M16 2v4"
            />
            <path
              d="M3 10h18"
            />
            <path
              d="M8 2v4"
            />
            <path
              d="M17 14h-6"
            />
            <path
              d="M13 18H7"
            />
            <path
              d="M7 14h.01"
            />
            <path
              d="M17 18h.01"
            />
          </svg>
          <span>
            S19
            : 
            11
             séances
          </span>
        </div>
        <button
          aria-label=" (désactivé)"
          class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed focus-enhanced border border-input bg-background shadow-sm hover:bg-accent hover:text-accent-foreground hover:shadow-md active:scale-95 h-8 rounded-md px-3 text-xs ml-auto md:ml-0"
          data-ui="refresh"
          disabled=""
          type="button"
        >
          <svg
            class="lucide lucide-refresh-ccw mr-2 h-4 w-4 animate-spin"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M21 12a9 9 0 0 0-9-9 9.75 9.75 0 0 0-6.74 2.74L3 8"
            />
            <path
              d="M3 3v5h5"
            />
            <path
              d="M3 12a9 9 0 0 0 9 9 9.75 9.75 0 0 0 6.74-2.74L21 16"
            />
            <path
              d="M16 16h5v5"
            />
          </svg>
          Actualiser
        </button>
      </div>
    </div>
    <div
      class="grid gap-6 lg:grid-cols-3"
    >
      <div
        class="rounded-xl border bg-card text-card-foreground shadow lg:col-span-2"
      >
        <div
          class="flex flex-col space-y-1.5 p-6 pb-0"
        >
          <div
            class="flex items-start justify-between gap-4"
          >
            <div>
              <h3
                class="font-semibold leading-none tracking-tight"
              >
                Évolution de l'humeur
              </h3>
              <p
                class="text-sm text-muted-foreground"
              >
                30 derniers jours, corrélation énergie & activités
              </p>
            </div>
            <button
              aria-label="Exporter l'évolution de l'humeur"
              class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed focus-enhanced hover:bg-accent hover:text-accent-foreground active:scale-95 h-9 w-9"
              type="button"
            >
              <svg
                class="lucide lucide-download h-4 w-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
                />
                <polyline
                  points="7 10 12 15 17 10"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="15"
                  y2="3"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="p-6 pt-0 space-y-6"
        >
          <div
            class="grid gap-4 sm:grid-cols-3"
          >
            <div>
              <p
                class="text-xs uppercase text-muted-foreground"
              >
                Humeur moyenne
              </p>
              <p
                class="text-2xl font-semibold"
              >
                7.7
                /10
              </p>
            </div>
            <div>
              <p
                class="text-xs uppercase text-muted-foreground"
              >
                Variation
              </p>
              <p
                class="text-2xl font-semibold text-emerald-600"
              >
                +
                0.8
              </p>
            </div>
            <div>
              <p
                class="text-xs uppercase text-muted-foreground"
              >
                Pic émotionnel
              </p>
              <p
                class="text-2xl font-semibold"
              >
                3 mai
              </p>
            </div>
          </div>
          <div
            class="h-[280px] w-full"
          >
            <div
              class="recharts-responsive-container"
              style="width: 100%; height: 100%; min-width: 0;"
            />
          </div>
        </div>
      </div>
      <div
        class="rounded-xl border bg-card text-card-foreground shadow"
      >
        <div
          class="flex flex-col space-y-1.5 p-6 pb-0"
        >
          <h3
            class="font-semibold leading-none tracking-tight"
          >
            Progression & Badges
          </h3>
          <p
            class="text-sm text-muted-foreground"
          >
            Cap sur le niveau suivant
          </p>
        </div>
        <div
          class="p-6 pt-0 space-y-6"
        >
          <div
            class="space-y-2"
          >
            <div
              class="flex items-center justify-between text-sm"
            >
              <span>
                Expérience
              </span>
              <span>
                1680
                 / 
                2200
              </span>
            </div>
            <div
              aria-label="Progression vers le niveau suivant"
              aria-valuemax="100"
              aria-valuemin="0"
              class="relative h-2 w-full overflow-hidden rounded-full bg-primary/20"
              data-max="100"
              data-state="indeterminate"
              role="progressbar"
            >
              <div
                class="h-full w-full flex-1 bg-primary transition-all"
                data-max="100"
                data-state="indeterminate"
                style="transform: translateX(-24%);"
              />
            </div>
            <p
              class="text-xs text-muted-foreground"
            >
              Encore 
              520
               pts avant le niveau 
              5
              .
            </p>
          </div>
          <div
            class="space-y-3"
          >
            <div
              class="flex items-start gap-3 rounded-lg border p-3"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-sparkles h-4 w-4 text-primary"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="m12 3-1.912 5.813a2 2 0 0 1-1.275 1.275L3 12l5.813 1.912a2 2 0 0 1 1.275 1.275L12 21l1.912-5.813a2 2 0 0 1 1.275-1.275L21 12l-5.813-1.912a2 2 0 0 1-1.275-1.275L12 3Z"
                />
                <path
                  d="M5 3v4"
                />
                <path
                  d="M19 17v4"
                />
                <path
                  d="M3 5h4"
                />
                <path
                  d="M17 19h4"
                />
              </svg>
              <div>
                <p
                  class="font-medium"
                >
                  Badge Sérénité
                </p>
                <p
                  class="text-xs text-muted-foreground"
                >
                  3 séances de respiration consécutives
                </p>
              </div>
            </div>
            <div
              class="flex items-start gap-3 rounded-lg border p-3"
            >
              <svg
                aria-hidden="true"
                class="lucide lucide-trending-up h-4 w-4 text-emerald-500"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <polyline
                  points="22 7 13.5 15.5 8.5 10.5 2 17"
                />
                <polyline
                  points="16 7 22 7 22 13"
                />
              </svg>
              <div>
                <p
                  class="font-medium"
                >
                  Boost hebdo
                </p>
                <p
                  class="text-xs text-muted-foreground"
                >
                  11
                   séances réalisées cette semaine
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="grid gap-6 lg:grid-cols-3"
    >
      <div
        class="rounded-xl border bg-card text-card-foreground shadow lg:col-span-2"
      >
        <div
          class="flex flex-col space-y-1.5 p-6 pb-0"
        >
          <div
            class="flex items-start justify-between gap-4"
          >
            <div>
              <h3
                class="font-semibold leading-none tracking-tight"
              >
                Séances par semaine
              </h3>
              <p
                class="text-sm text-muted-foreground"
              >
                Répartition guidée, respiration, VR et journal
              </p>
            </div>
            <button
              aria-label="Exporter les sessions hebdomadaires"
              class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed focus-enhanced hover:bg-accent hover:text-accent-foreground active:scale-95 h-9 w-9"
              type="button"
            >
              <svg
                class="lucide lucide-download h-4 w-4"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
                />
                <polyline
                  points="7 10 12 15 17 10"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="15"
                  y2="3"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="p-6 pt-0 space-y-6"
        >
          <div
            class="grid gap-4 sm:grid-cols-3"
          >
            <div>
              <p
                class="text-xs uppercase text-muted-foreground"
              >
                Moyenne
              </p>
              <p
                class="text-2xl font-semibold"
              >
                9.5
                 séances
              </p>
            </div>
            <div>
              <p
                class="text-xs uppercase text-muted-foreground"
              >
                Semaine max
              </p>
              <p
                class="text-2xl font-semibold"
              >
                S19
              </p>
            </div>
            <div>
              <p
                class="text-xs uppercase text-muted-foreground"
              >
                Guidées
              </p>
              <p
                class="text-2xl font-semibold"
              >
                4
              </p>
            </div>
          </div>
          <div
            class="h-[280px] w-full"
          >
            <div
              class="recharts-responsive-container"
              style="width: 100%; height: 100%; min-width: 0;"
            />
          </div>
        </div>
      </div>
      <div
        class="rounded-xl border bg-card text-card-foreground shadow"
      >
        <div
          class="flex flex-col space-y-1.5 p-6 pb-0"
        >
          <h3
            class="font-semibold leading-none tracking-tight"
          >
            Focus de la semaine
          </h3>
          <p
            class="text-sm text-muted-foreground"
          >
            Moments les plus vibrants
          </p>
        </div>
        <div
          class="p-6 pt-0 space-y-4"
        >
          <div
            class="rounded-lg border bg-muted/40 p-4"
          >
            <p
              class="text-sm text-muted-foreground"
            >
              Créneau le plus intense
            </p>
            <p
              class="text-lg font-semibold"
            >
              Jeu
               — 
              Soir
            </p>
            <p
              class="text-sm text-muted-foreground"
            >
              2
               
              séances
              , vibe 
              serein
            </p>
          </div>
          <div
            class="space-y-3 text-sm"
          >
            <div
              class="flex items-center justify-between"
            >
              <span>
                Pic d'énergie
              </span>
              <span>
                3 mai
              </span>
            </div>
            <div
              class="flex items-center justify-between"
            >
              <span>
                Sessions VR cumulées
              </span>
              <span>
                3
              </span>
            </div>
            <div
              class="flex items-center justify-between"
            >
              <span>
                Respirations guidées
              </span>
              <span>
                5
              </span>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="rounded-xl border bg-card text-card-foreground shadow"
    >
      <div
        class="flex flex-col space-y-1.5 p-6 pb-0"
      >
        <div
          class="flex items-start justify-between gap-4"
        >
          <div>
            <h3
              class="font-semibold leading-none tracking-tight"
            >
              Heatmap Vibes
            </h3>
            <p
              class="text-sm text-muted-foreground"
            >
              Intensité émotionnelle par moment clé de la journée
            </p>
          </div>
          <button
            aria-label="Exporter la heatmap des vibes"
            class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium transition-all duration-200 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 disabled:cursor-not-allowed focus-enhanced hover:bg-accent hover:text-accent-foreground active:scale-95 h-9 w-9"
            type="button"
          >
            <svg
              class="lucide lucide-download h-4 w-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M21 15v4a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2v-4"
              />
              <polyline
                points="7 10 12 15 17 10"
              />
              <line
                x1="12"
                x2="12"
                y1="15"
                y2="3"
              />
            </svg>
          </button>
        </div>
      </div>
      <div
        class="p-6 pt-0 space-y-6"
      >
        <div
          class="h-[320px] w-full"
        >
          <div
            class="recharts-responsive-container"
            style="width: 100%; height: 100%; min-width: 0;"
          />
        </div>
        <div
          class="flex flex-wrap items-center gap-3 text-xs text-muted-foreground"
        >
          <div
            class="flex items-center gap-2"
          >
            <span
              class="h-3 w-6 rounded-full"
              style="background-color: rgba(124, 58, 237, 0.95);"
            />
            <span>
              Intense
            </span>
          </div>
          <div
            class="flex items-center gap-2"
          >
            <span
              class="h-3 w-6 rounded-full"
              style="background-color: rgba(139, 92, 246, 0.85);"
            />
            <span>
              Élevé
            </span>
          </div>
          <div
            class="flex items-center gap-2"
          >
            <span
              class="h-3 w-6 rounded-full"
              style="background-color: rgba(167, 139, 250, 0.75);"
            />
            <span>
              Modéré
            </span>
          </div>
          <div
            class="flex items-center gap-2"
          >
            <span
              class="h-3 w-6 rounded-full"
              style="background-color: rgba(229, 231, 235, 0.45);"
            />
            <span>
              Calme
            </span>
          </div>
          <div
            class="flex items-center gap-2"
          >
            <span
              class="h-3 w-6 rounded-full"
              style="background-color: rgba(229, 231, 235, 0.45);"
            />
            <span>
              Repos
            </span>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
